R/safe2x2Test.R

```diff
@@ -1,1148 +1,2016 @@
-    cat(paste0(n, " "))
-
-    #if na = nb and we are testing twosided we only need to test the delta ump,
-    #since this is then known analytically
-
-    if (na == nb & alternative == "two.sided" & any(is.na(priorInfo))) {
-      delta.ump <- get_delta_ump_analytically(n = na + nb, alpha = alpha,
-                                              lower = 0.1, upper = 0.9)
-
-      if (!is.na(delta.ump)) {
-        deltas <- delta.ump
-      }
-    }
+#' @rdname safeTwoProportionsTest
+#'
+#' @export
+safe.proportion.test <- function(ya, yb, designObj = NULL, pilot = FALSE) {
+  res <- tryCatch(safeTwoProportionsTest(ya = ya, yb = yb, designObj = designObj, pilot = pilot),
+                  error = function(e){e})
 
-    #start iterating over possible delta stars at this n
-    for (j in 1:length(deltas)) {
-      delta <- deltas[j]
-
-      if(!any(is.na(priorInfo))){
-        #prior knowledge on theta_a: construct simple S
-        H1set <- matrix(c(priorInfo[["h0"]], priorInfo[["RR"]]*priorInfo[["h0"]]),
-                        nrow = 1)
-        point_h0 <- (na*H1set[1,1]+nb*H1set[1,2])/(na+nb)
-        w1 <- 1
-      } else {
-        #construct parameters to use in the S-test
-        if (alternative == "two.sided") {
-          if (na == nb) {
-            H1set <- create_h1_set_equal_group_sizes(delta)
-            w1 <- c(0.5, 0.5)
-            point_h0 <- 0.5
-          } else {
-            H1set <- create_h1_set_unequal_group_sizes(delta, na = na, nb = nb)
-            w1 <- c(0.5, 0.5)
-            point_h0 <- 0.5
-          }
-        } else {
-          #if alternative is greater, feed negative delta to create point h1 and h0
-          delta_design <- ifelse(alternative == "greater", -delta, delta)
-          point_H1_and_H0 <- create_point_h1_and_h0_one_sided(delta = delta_design, na = na, nb = nb)
-
-          if (any(is.na(point_H1_and_H0))) {
-            next
-          }
-
-          H1set <- point_H1_and_H0$H1set
-          w1 <- 1
-          point_h0 <- point_H1_and_H0$point_h0
-        }
-      }
-
-      #check if parameters valid, between 0 and 1
-      if(any(H1set <= 0) | any(H1set >=  1)){
-        next
-      }
-
-      percent.reject.S <-
-        perform_MC_simulations_for_each_data_generating_distribution(H1.deltamin = H1DeltaMin, na = na, nb = nb,
-                                                                     seed.time = seedTime, M = M, H1set = H1set,
-                                                                     w1 = w1, point_h0 = point_h0, alpha = alpha)
-
-      #check if desired power has been reached at desired precision
-      if (min(percent.reject.S) >= (1 - beta)) {
-        notreached <- FALSE
-
-        cat("\nFor all p1, power above desired level. Worst case: ")
-        cat(min(percent.reject.S))
-        cat(" with data generated from: ")
-        cat(H1DeltaMin[which.min(percent.reject.S),]); cat(".\n")
-        break
-      }
-    }
+  if(!is.null(res[["message"]])){
```

Comment:
ik heb een functie geschreven `isTryError` welke je misschien kunt gebruiken en `tryOrFailWithNA`

Reply:
jullie gebruiken verder in safeS3methods overal al is.null voor lijsten, dat maar consequent blijven doen? alles veranderen een keer in volgende versie?

Reply:
Ik wil specifiek de message van de error teruggeven, dus dit is makkelijker